### PR TITLE
Junit 5 support.

### PR DESCRIPTION
**Pull request by felixdesouza**
Surprised no one has needed this for JUnit5. Got bored yesterday so just hashed it out in a similar manner to how https://github.com/palantir/docker-compose-rule did it.

Annoying bit for me is the integration tests, I blindly copied them since one uses junit4 and the other uses junit jupiter, and I couldn't think of a way to share the tests in a meaningful manner. Happy to trim down the tests to some smaller subset that you think is useful.

The main package is still there, so people won't be immediately broken, and they'll still pull in the junit4 version. For the most part, as long as this isn't wrapped and reexported e.g. `DockerComposeRule`, then it's just a binary incompatible change which doesn't actually matter since we'll just recompile. To err on the safe side, we could just 1.0. Versions are basically free for libraries right? 😅

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -17,44 +17,37 @@ buildscript {
...
-targetCompatibility = 1.8
+configure(subprojects) {
+    apply plugin: 'java-library'
+    apply plugin: 'org.inferred.processors'
 
-dependencies {
-    compile group: 'com.palantir.docker.compose', name: 'docker-compose-rule-junit4'
-    compile group: 'junit', name: 'junit'
-    compile group: 'one.util', name: 'streamex'
+    tasks.check.dependsOn(javadoc)
```
Comment (SerialVelocity) [inline build.gradle]: Why is this needed?

File: docker-proxy-junit-jupiter/src/main/java/com/palantir/docker/proxy/DockerProxyExtension.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     * Creates a {@link DockerProxyExtension} which will create a proxy and DNS so that tests can interface with docker
+     * containers directly.
+     *
+     * @param dockerContainerInfoCreator A {@link Function} that creates the DockerContainerInfo to use
+     * @param classToLogFor              The class using {@link DockerProxyExtension}
+     */
+    DockerProxyExtension(
+            Function<DockerExecutable, DockerContainerInfo> dockerContainerInfoCreator,
+            Class<?> classToLogFor) {
+        super(customizer -> customizer.apply(DockerComposeExtension.builder()).build(),
```
Comment (SerialVelocity) [inline docker-proxy-junit-jupiter/src/main/java/com/palantir/docker/proxy/DockerProxyExtension.java]: `.retryAttempts(0)` here?

File: build.gradle (hunk 1 above)
Comment (felixdesouza) [reply build.gradle]: it's not, can remove!

File: docker-proxy-junit-jupiter/src/main/java/com/palantir/docker/proxy/DockerProxyExtension.java (hunk 2 above)
Comment (felixdesouza) [reply docker-proxy-junit-jupiter/src/main/java/com/palantir/docker/proxy/DockerProxyExtension.java]: since we're using the generic builder, it doesn't have `retryAttempts`, it's pretty sad

Comment (SerialVelocity) [reply docker-proxy-junit-jupiter/src/main/java/com/palantir/docker/proxy/DockerProxyExtension.java]: Why does the generic builder matter here? In the other place you did `DCR.builder().retryAttempts(0)`. Can't you just do `DCE.builder().retryAttempts(0)` here?

Comment (felixdesouza) [reply docker-proxy-junit-jupiter/src/main/java/com/palantir/docker/proxy/DockerProxyExtension.java]: sorry, I responded to the wrong comment, there was a comment, I think it's disappeared, where you were asking why I had to put `retryAttempts` in a particular place.

**Review by SerialVelocity — APPROVED** [review]

**Comment by felixdesouza** [comment]
💃
